### PR TITLE
Add parseNativeToWrapped

### LINKED ISSUE #264: Create HandleTokenManager parseNativeToWrapped and deprecate checkForHlpNativeToken
1. Create a new method 'parseNativeToWrapped' which accepts any token and if it is native, parses it to the wrapped native token.

2. Delete checkForHlpNativeToken and replace all instances of it with the new 'parseNativeToWrapped' method

### PR DESCRIPTION
**Pull request by Preston-Harrison**
resolves #264

### CODE REVIEW

**Comment by Preston-Harrison** [comment]
sonarcloud is mad because I am using a deprecated function in a test

File: src/components/TokenManager/HandleTokenManager.ts [hunk 1]
```diff
@@ -176,6 +189,33 @@ class HandleTokenManager extends TokenManager {
...
+
+  /**
+   * Parses a token into a wrapped native token if it is native
+   * @param token the token to parse
+   * @returns an object with the properties isInputNative and parsedToken. If the token passed into the
+   * function is native, isInputNative will be set to true, and parsedToken will be the native wrapped token
+   * for the network. If the token passed in is not native, isInputNative will be set to false and parsedToken
+   * will be the token passed into the function
+   */
+  public parseNativeToWrapped(token: TokenInfo): {
```
Comment (brettvniekerk) [inline src/components/TokenManager/HandleTokenManager.ts:201]: why not just return the wrapped token or `undefined`?

Comment (brettvniekerk) [reply src/components/TokenManager/HandleTokenManager.ts:201]: if `undefined` then the calling function knows that the token isn't native

File: src/components/TokenManager/HandleTokenManager.ts [hunk 2]
```diff
@@ -176,6 +189,33 @@ class HandleTokenManager extends TokenManager {
...
+  } {
+    if (!token.extensions?.isNative) {
+      return {
+        isInputNative: false,
+        parsedToken: token,
+      };
+    }
+
+    const wrappedNative = this.getWrappedNativeToken(token.chainId);
+    if (!wrappedNative) throw new Error("");
```
Comment (brettvniekerk) [inline src/components/TokenManager/HandleTokenManager.ts:213]: no error message?

File: src/components/TokenManager/index.ts [hunk 3]
```diff
@@ -160,6 +160,17 @@ class TokenManager {
     );
   }
 
+  /**
+   * Finds the wrapped native token for a network
+   * @param network the network from which to get the token
+   * @returns the wrapped native token if one exists, otherwise undefined
+   */
+  public getWrappedNativeToken(network: Network | number) {
```
Comment (brettvniekerk) [inline src/components/TokenManager/index.ts:168]: this and `parseNativeToWrapped` seem to do the same thing

File: src/config/TokenLists/misc-tokens.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "Miscellaneous Tokens",
+  "timestamp": "2022-05-18T01:09:39+0000",
+  "version": {
```
Comment (brettvniekerk) [inline src/config/TokenLists/misc-tokens.json:4]: `1.0.0`?

**Review by brettvniekerk — CHANGES_REQUESTED** [review]
few things to address

File: src/config/TokenLists/misc-tokens.json (hunk 4 above)
Comment (Preston-Harrison) [reply src/config/TokenLists/misc-tokens.json:4]: yes

File: src/components/TokenManager/index.ts (hunk 3 above)
Comment (Preston-Harrison) [reply src/components/TokenManager/index.ts:168]: getWrappedNativeToken gets the wrapped native token. parseNativeToWrapped returns the wrapped native token only if the input token is native

File: src/components/TokenManager/HandleTokenManager.ts (hunk 2 above)
Comment (Preston-Harrison) [reply src/components/TokenManager/HandleTokenManager.ts:213]: lol

File: src/components/TokenManager/HandleTokenManager.ts (hunk 1 above)
Comment (Preston-Harrison) [reply src/components/TokenManager/HandleTokenManager.ts:201]: The purpose of this function is to be similar to the utility function checkForHlpNativeToken, but not throw if the input token is not an hLP token. I'll admit the output is a bit strange, but it serves a common purpose really well. See below.

```javascript
const { isInputNative, parsedToken } = tokenManager.parseNativeToWrapped(possiblyNativeToken);
if (isInputNative) {
    contract.doSomethingEth(parsedToken.address, { value: someValue });
} else {
    contract.doSomething(parsedToken.address, someValue);
}
```

I don't think it should return undefined, as that is not common for handling an invalid input into a parsing function. if anything it should throw if the input is not a native token. But then it gets really annoying as you have to wrap the thing in a try-catch.

Comment (brettvniekerk) [reply src/components/TokenManager/HandleTokenManager.ts:201]: i dont like how both properties essentially reflect the same thing - it seems like an unnecessary doubling-up of data. i would rather you use a utility function or refactor to use one return type - but if it is common for parsing functions then dont feel the need

File: src/components/TokenManager/index.ts (hunk 3 above)
Comment (brettvniekerk) [reply src/components/TokenManager/index.ts:168]: ah, nevermind then

File: src/components/TokenManager/HandleTokenManager.ts (hunk 1 above)
Comment (Preston-Harrison) [reply src/components/TokenManager/HandleTokenManager.ts:201]: They don't reflect the same thing when the token passed in is WETH

File: src/components/TokenManager/HandleTokenManager.ts [hunk 5]
```diff
@@ -79,7 +91,7 @@ class HandleTokenManager extends TokenManager {
     address: string,
     network: Network | number
   ): boolean {
-    const tokens = this.getLoadedTokens(network);
+    const tokens = this.getHlpTokens(network);
```
Comment (rellfy) [inline src/components/TokenManager/HandleTokenManager.ts:94]: All these hLP token getter functions that were `getLoadedTokens` and filtering by `isHlpToken` themselves. Now that they are getting tokens by `getHlpTokens`, they should no longer filter by `isHlpToken` as they're still doing

Comment (Preston-Harrison) [reply src/components/TokenManager/HandleTokenManager.ts:94]: good point

Comment (Preston-Harrison) [reply src/components/TokenManager/HandleTokenManager.ts:92]: fixed

**Comment by Preston-Harrison** [comment]
sonarcloud errors can be ignored

**Review by rellfy — APPROVED** [review]
Do you need to increase the package version before merging?

**Comment by Preston-Harrison** [comment]
> Do you need to increase the package version before merging?

no, we can release it next time

File: src/components/TokenManager/HandleTokenManager.ts [hunk 6]
```diff
@@ -48,9 +56,13 @@ class HandleTokenManager extends TokenManager {
    * @returns all hLP tokens for the network
    */
   public getHlpTokens(network: number | Network): TokenInfo[] {
-    return this.getLoadedTokens(network).filter(
-      (token) => token.extensions?.isHlpToken
+    const tokens = this.getFromCache(HANDLE_TOKEN_KEY)?.tokens.filter((t) =>
```
Comment (brettvniekerk) [inline src/components/TokenManager/HandleTokenManager.ts:59]: no single letter vars, please call `token`

**Review by brettvniekerk — APPROVED** [review]
just one thing to look at, otherwise LGTM

File: src/components/TokenManager/HandleTokenManager.ts (hunk 6 above)
Comment (Preston-Harrison) [reply src/components/TokenManager/HandleTokenManager.ts:59]: not a big deal here, meaning is obvious IMO

**Comment by Preston-Harrison** [comment]
force merging as coverage is failing due to oracles

**Pull request merged**